src/Internal/Payment/State/InitialState.php

```diff
@@ -109,6 +121,13 @@ public function start_processing( PaymentRequest $request ) {
 		// Start multiple verification checks.
 		$this->process_order_phone_number();
 
+		if ( $this->fraud_prevention_service->is_enabled()
+			&& ! $this->fraud_prevention_service->verify_token( $this->get_context()->get_fraud_prevention_token() ) ) {
+			throw new PaymentProcessingException(
```

Comment:
Alright, you got me here. Let's use `StateTransitionException` for now, to avoid adding a new exception class. We'll do better with @zmaglica in https://github.com/Automattic/woocommerce-payments/issues/7425.

Reply:
Resolved a50c4a40993a7929e7e0c16e31b655eca82aaaa6